usr/src/cmd/mailx/cmd1.c

```diff
@@ -238,19 +238,19 @@ printhead(int mesg)
 	showto = 0;
 	if ((mp->m_flag & (MREAD|MNEW)) == (MREAD|MNEW))
 		dispc = 'R';
-	if (!(int)value("bsdcompat") && (mp->m_flag & (MREAD|MNEW)) == MREAD)
+	if (!value("bsdcompat") && (mp->m_flag & (MREAD|MNEW)) == MREAD)
```

Comment:
```suggestion
	if (value("bsdcompat") == NULL && (mp->m_flag & (MREAD|MNEW)) == MREAD)
```
Since `value()` returns a `char *` this should be an explicit comparison, and throughout.
The original (cast to int then boolean comparison) is awful!